src/libs/antares/study/parts/short-term-storage/series.cpp

```diff
@@ -144,4 +115,46 @@ void Series::fillDefaultSeriesIfEmpty()
     fillIfEmpty(upperRuleCurve, 1.0);
 }
 
+bool Series::validate() const
+{
+    if (maxInjectionModulation.size() != HOURS_PER_YEAR
+        || maxWithdrawalModulation.size() != HOURS_PER_YEAR || inflows.size() != HOURS_PER_YEAR
+        || lowerRuleCurve.size() != HOURS_PER_YEAR || upperRuleCurve.size() != HOURS_PER_YEAR)
+    {
+        logs.warning() << "Size of series for short term storage is wrong";
+        return false;
+    }
+
+    auto checkVectBetween = [](const std::vector<double>& v) {
```

Comment:
About lambda function **checkVectBetween** : 
To me, it should be renamed **checkTimeSeriesBetweenBounds**, and the bounds should appear as arguments : 
```C++
auto checkTimeSeriesBetweenBounds= [](const std::vector<double>& ts, double min, double max) {
    return std::all_of(v.begin(), v.end(), [](double d){ return (d >= min && d <= max); });
    };
```